src/app/App.tsx

```diff
@@ -59,11 +59,19 @@ PortalParent.displayName = 'PortalParent';
 const ModalPortalParent = () => <PortalParent ref={modalPortalRef} />;
 const GlobalLoaderParent = () => <PortalParent ref={loaderPortalRef} />;
 
-const App = ({ children, config }: { children: ReactNode; config: any }) => (
+const App = ({
+	children,
+	config,
+	jwt,
+}: {
+	children: ReactNode;
+	config: any;
+	jwt?: typeof document.cookie;
+}) => (
 	<ThemeProvider>
 		<QueryClientProvider client={queryClient}>
 			<AppProvider config={config}>
-				<AuthProvider>
+				<AuthProvider jwt={jwt}>
```

Comment:
as a segue to my previous comment (and I see now Ciaran asked about this in his review), it doesn't look like any parents of the `AuthProvider` have any use or need for the jwt, so it would be best to relocate the cookie getter to the context itself :cookie:

Reply:
cc @ciaranschutte 
In principle, yes. The problem here is hydration between client + server. See: https://github.com/icgc-argo/rdpc-ui/blob/develop/src/app/App.tsx#L19

We pre-render the page, but there are cookies in the browser. So the pre-render and the browser render don't match, and the AuthProvider does not have a cookie server side, because there is none on the initial server request.

That's why I was using Cookie in layout; App and AuthProvider are 'use client'

Reply:
oh! I see! sounds like a good case for having only part of the auth handling be `client` side e.g. the react context, I imagine (my "what's allowed server-side"-fu is not strong enough). 

My concern here is how the auth stuff may start bleeding out of its scope, breaking the separation of concerns principle. Is it possible to have a server-side pre-render (or whatever the right term is) for the auth handlers?